CommonMark/Formatters/HtmlFormatter.cs

```diff
@@ -495,6 +497,40 @@ protected virtual void WriteInline(Inline inline, bool isOpening, bool isClosing
                     }
                     break;
 
+                case InlineTag.Placeholder:
+                    ignoreChildNodes = false;
+
+                    if (isOpening)
+                    {
+                        string placeholderSubstitute = (_placeholderResolver != null) ? _placeholderResolver(inline.TargetUrl) : null;
+
+                        if (placeholderSubstitute != null)
+                        {
+                            ignoreChildNodes = true;
+
+                            if (Settings.TrackSourcePosition) WritePositionAttribute(inline);
+                            Write(placeholderSubstitute);
+                            _endPlaceholders.Push(null);
+                        }
+                        else
+                        {
+                            ignoreChildNodes = false;
+                            Write("[");
+                            _endPlaceholders.Push(']');
+                        }
+                    }
+                    if (isClosing)
+                    {
+                        var closingChar = _endPlaceholders.Pop();
+
+                        ignoreChildNodes = !closingChar.HasValue;
```

Comment:
ignoreChildNodes on the isClosing branch should have no effect and can be ignored